pkg/kubelet/kubelet_pods.go

```diff
@@ -1269,6 +1269,15 @@ func getPhase(spec *v1.PodSpec, info []v1.ContainerStatus) v1.PodPhase {
 	stopped := 0
 	succeeded := 0
 	for _, container := range spec.Containers {
+
+		//exclude sidecars from pod phase calculations
+		if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+			if container.Lifecycle != nil && container.Lifecycle.Type != nil {
+				if *container.Lifecycle.Type == v1.LifecycleTypeSidecar {
```

Comment:
if we always skip sidecar, pod will still be pending if we only started sidecar, which is yet another behavior change. Pls make sure the community is ok with it and we modify the doc https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase

Reply:
That's not really a change to the definition, Pending =
> The Pod has been accepted by the Kubernetes system, but one or more of the Container images has not been created. This includes time before being scheduled as well as time spent downloading images over the network, which could take a while.

But yeah it's definitely worth noting and calling out

Reply:
semantically I'd prefer we keep the "PodRunning" statement, if sidecars started to get created / running, we shall mark pod as running, as "pod running" shall _not_ be equivalent to "app / user logic running", which seems to be what we are trying to implement here